packages/coachmark/src/Coachmark.ts

```diff
@@ -13,32 +13,346 @@ governing permissions and limitations under the License.
+    // event on secondary button
+    private handleSecondaryCTA(): void {
+        this.dispatchEvent(
+            new Event('secondary', {
+                bubbles: true,
+                cancelable: true,
+                composed: true,
+            })
+        );
+        this.close();
+    }
+
+    protected renderButtons(): TemplateResult {
+        return html`
+            <sp-button-group class="spectrum-ButtonGroup buttongroup">
+                ${when(
+                    this.secondaryCTA,
+                    () => html`
+                        <sp-button
+                            treatment="outline"
+                            variant="secondary"
+                            @click=${this.handleSecondaryCTA}
+                        >
+                            ${this.secondaryCTA}
+                        </sp-button>
+                    `
+                )}
+                ${when(
+                    this.primaryCTA,
+                    () => html`
+                        <sp-button
+                            size="s"
+                            treatment="outline"
+                            variant="primary"
+                            @click=${this.handlePrimaryCTA}
+                        >
+                            ${this.primaryCTA}
+                        </sp-button>
+                    `
+                )}
+            </sp-button-group>
+            <sp-button-group
+                class="spectrum-ButtonGroup buttongroup-mobile"
+                size="s"
+            >
+                ${when(
+                    this.secondaryCTA,
+                    () => html`
+                        <sp-button
+                            variant="secondary"
+                            treatment="outline"
+                            icon-only
+                            aria-label="previous"
+                            @click=${this.handleSecondaryCTA}
+                        >
+                            <sp-icon-chevron200
+                                size="s"
+                                class="spectrum-UIIcon-ChevronLeft75"
+                                slot="icon"
+                            ></sp-icon-chevron200>
+                        </sp-button>
+                    `
+                )}
+                ${when(
+                    this.primaryCTA,
+                    () => html`
+                        <sp-button
+                            size="s"
+                            treatment="outline"
+                            variant="primary"
+                            @click=${this.handlePrimaryCTA}
+                        >
+                            ${this.primaryCTA}
+                        </sp-button>
+                    `
+                )}
+            </sp-button-group>
+        `;
+    }
 
-    @property({ reflect: true })
-    public variant: 'dark' | 'light' | '' = '';
+    // render steps
+    protected renderSteps = (): TemplateResult => {
```

Comment:
```suggestion
    protected renderSteps(): TemplateResult => {
```

This is an anonymous render function, which could have negative performance implication.

Reply:
We have to use an arrow function since this will become undefined here when we use when directive

Reply:
I see. We can keep it this way for now but I'm going to make an issue for it to remind us to test the performance of this area later.

Reply:
https://github.com/adobe/spectrum-web-components/issues/3723